lib/src/repo.rs

```diff
@@ -606,8 +606,16 @@ impl MutableRepo {
         Ok(rebaser.rebased().len())
     }
 
-    pub fn set_wc_commit(&mut self, workspace_id: WorkspaceId, commit_id: CommitId) {
+    pub fn set_wc_commit(
+        &mut self,
+        workspace_id: WorkspaceId,
+        commit_id: CommitId,
+    ) -> Result<(), OpenedRootCommit> {
+        if &commit_id == self.store().root_commit_id() {
+            return Err(OpenedRootCommit);
+        }
```

Comment:
I'd have liked to do this inside `View` for even more robustness, but I couldn't find an easy way to get the root commit ID there.

Reply:
The `View` object is a pretty low-level data object and the root commit ID can vary between commit backends, so that's why it's not available there. Doing it here in `MutableRepo` sounds good to me.